pkg/repro/repro.go

```diff
@@ -168,37 +171,54 @@ func (ctx *context) repro(entries []*prog.LogEntry, crashStart int) (*Result, er
 
 	res, err := ctx.extractProg(entries)
 	if err != nil {
-		return res, err
+		return nil, err
 	}
 	if res == nil {
 		return nil, nil
 	}
-
+	defer func() {
+		res.Opts.Repro = false
+	}()
 	res, err = ctx.minimizeProg(res)
 	if err != nil {
-		return res, err
+		return nil, err
 	}
 
+	// Try extracting C repro without simplifying options first.
+	res, err = ctx.extractC(res)
+	if err != nil {
+		return nil, err
+	}
+	if res.CRepro {
+		goto have_c
+	}
+
+	// Simplify options.
 	res, err = ctx.simplifyProg(res)
 	if err != nil {
-		return res, err
+		return nil, err
+	}
+	if res.CRepro {
+		// We try extracting C repros after each simplification step in simplifyProg().
+		goto have_c
 	}
 
+	// Try extracting C repro with options simplified.
```

Comment:
LInes 206-213 are excessive, we already tried to extract C right before we get here. So you can just drop lines 201-210.
And then we can further simplify lines 187-220 with:

```
	res, err = ctx.extractC(res)
	if err != nil {
		return nil, err
	}
	if !res.CRepro {
		res, err = ctx.simplifyProg(res)
		if err != nil {
			return nil, err
		}
	}
	if res.CRepro {
		res, err = ctx.simplifyC(res)
		if err != nil {
			return nil, err
		}
	}
```